tests/Representation.Tests/TestUtils/MockArguments.cs

```diff
@@ -0,0 +1,18 @@
+﻿using Moq;
+using Symbolica.Abstraction;
+
+namespace Symbolica.Representation.TestUtils
+{
+    internal static class MockArguments
+    {
+        public static IArguments Create(params Nibble[] arguments)
+        {
+            var mockArguments = new Mock<IArguments>(MockBehavior.Strict);
```

Comment:
I tried it both ways and definitely prefer the mock way for this. I use the stub/fake way of doing things in the `Computation.Tests` project so I am not particularly tied to either one. It just came out cleaner, shorter and more understandable (to me) for these specific tests. The issue I had is that fully implementing `IState` and `IExpression` is actually pretty involved (3 of the other projects are doing exactly this). I realise this may suggest that these interfaces are doing too much, but I am generally pretty happy with the design overall (and I haven't found a better one yet). Obviously I could just use `NotImplementedException`s or something but there are really a lot of them and it comes out pretty noisy. Horses for courses and this was my favourite horse for this one.

Reply:
I suppose the short answer is that although the `Nibble` type is quite long and weird, it is really just boilerplate, and is the right level of abstraction to make the actual tests, which are much trickier, as easy as possible to follow. I really wanted to use `BitArray` rather than anything numeric but it doesn't have everything I need.